### PR TITLE
implemented SNICAR SSP table internally in Icepack

### PR DESCRIPTION
**Pull request by apcraig**
This implements the SNICAR SSP table internally in Icepack rather than having it be read in.  This is a prototype.  It seems to be bit-for-bit with the table reading version.  If we move forward, we'd want to decide whether to

- keep ability to initialize SNICAR SSP data via calls to icepack_parameters, but probably remove "snicar" option and formalize "file" option.
- think about moving around location of some module data
- update the snw_ssp_table options.  in this example, the options are 'test', 'snicar_from_file', 'snicardata_table'.  those are not the best names.

The new file is icepack_snicardata.F90.  The core of this file was created primarily via a script that ncdumps the netcdf file and then parses it into Fortran.  We could think about archiving this script somewhere.

One negative is that it takes a relatively long time (O(30 seconds)) to compile icepack_snicardata.F90.  I guess this is because it's pretty big.  I think that's OK for general use, but for Icepack suite testing, we may want to implement some of tricks we've done in CICE like reuse binaries and so forth.  Ultimately, the extra (largely minor) compile time is probably still beneficial overall.

### CODE REVIEW

File: configuration/scripts/options/set_nml.snicardata [hunk 1]
```diff
@@ -0,0 +1,2 @@
+  shortwave          = 'dEdd_snicar'
+  snw_ssp_table      = 'snicardata_table'
```
Comment (eclare108213) [inline configuration/scripts/options/set_nml.snicardata:2]: and rename this one as set_nml.snicartable?

**Review by eclare108213 — COMMENTED** [review]
This looks okay to me, although I can't yet see the new module with all of the data in it.  My suggestions below are to be consistent in the naming.  Maybe this isn't the best choice, but I went with
*file read 1471 data from the netcdf file
*table use 1471 data from the new module
*test use the 5x5 data

We could implement both options in CICE and just the 'table' option in Icepack, or the other way around.  My personal preference might be to get rid of the netcdf file altogether, since the metadata is incomplete.  It would be good to store the original file and the script used to parse it somewhere, not sure where.

**Comment by apcraig** [comment]
I agree we need better names for variables, settings, and namelist ultimately.  But first we should decide what we want in the bigger picture.  Do we want the hardwired table at all?  Do we want to keep the ability to read the current netcdf file?  That is partly hardcoded to "1471".  

Then how does reading the current file play into a possibility to read a different size file in the future and the code's ability to be flexible about the various data array sizes (i.e. 5, 1471, nmodal1/2, etc)?  It would be nice if all the data needed, including the bands and grain sizes were ALL in the netcdf file and not necessarily hardwired.  That would give us more flexibility to read a new file in the future.  We also might want to think about making the interpolation more flexible instead of just hardwiring the index table based on knowing the spectral bands and grain sizes.

This is what I think.  We should drop the ability to set the data via reading the current file because
- we don't need it if we have the hardwired table.
- the current implementation does not provide enough flexibility (wrt reading the spectral bands or grain sizes AND wrt interpolation)
- but we consider keeping the ability to set the data via icepack_parameters to provide some future extensibility, but we'd also add the spectral bands and grain sizes as arguments.   And we'd add some aborts that would not allow an arbitrary set of tables to run in Icepack until we fix the interpolation.

So the options might be "test", "snicar", and "input" with "snicar" being the hardwired tabled and "input" supporting ability to pass in data but otherwise disabled.  Or something like that.

Finally, just FYI, you can see the new big file if you click thru to "view file" on the 3 right dots then "view raw".

**Comment by apcraig** [comment]
Also, I think we should have a short discussion to include the rest of the team (including Nicole and Dave for instance) to see what they think of file vs hardcoded just to get additional feedback.  Do we know how MPASSI sets the data inside the columnphysics and if there was a particular reason that approach was chosen?

**Comment by eclare108213** [comment]
I agree with all of your points above, except that I tried 'view file' under the 3 dots and github said 'no, sorry'.  I was thinking that we (on behalf of the consortium) could opt to reject the netcdf file because of its deficiencies, as long as we have an approach that is as good or better.  

I don't know the answer to your MPAS-SI question, but I suspect that it's in a file simply because it's big, not because there's any intent to provide alternative data at some point in the future. We can certainly talk about this first thing tomorrow.  

I do agree that we could consider keeping this data in icepack_parameters, but in that case we should treat all of the data the same, i.e. have none of it stored in icepack_shortwave.F90.  Conceptually, I think it belongs in the shortwave module, not namelist/forcing.

**Comment by apcraig** [comment]
See #9

**Pull request closed without merging**